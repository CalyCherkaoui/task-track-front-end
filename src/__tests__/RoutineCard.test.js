import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store';
import RoutineCard from '../component/RoutineCard';
import routineFixture from '../fixtures/routineFixt';

describe(
  'Renders correctly Routine card component',
  () => {
    let container;
    beforeEach(() => {
      container = shallow(
        <Provider store={store}>
          <Router>
            <RoutineCard routine={routineFixture.data.attributes} />
          </Router>
        </Provider>,
      );
    });

    it(
      'Contains Routine\'s name',
      () => {
        expect(container.find(routineFixture.data.attributes.name)).toBeTruthy();
      },
    );

    it(
      'Matches Routine Card snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <RoutineCard routine={routineFixture.data.attributes} />
            </Router>
          </Provider>,
        );

        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
