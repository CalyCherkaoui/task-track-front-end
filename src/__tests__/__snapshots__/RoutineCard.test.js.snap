// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly Routine card component Matches Routine Card snapshot 1`] = `
<div
  className="card_wrapper"
>
  <div
    className="card_icon"
  >
    <i
      className="GiBookmarklet fa-2x"
    />
  </div>
  <div
    className="card_title"
  >
    tech-reading
  </div>
  <div
    className="card_title"
  >
    2
  </div>
</div>
`;
